README.md

```diff
@@ -1,2 +1,193 @@
 # ast-gh-actions
+
 A library or collection of AST github shared composite actions.
+
+## Development
+
+### Git Pre-commit Hook
+
+The repository uses a pre-commit hook to run linting on all TypeScript projects. To enable:
+
+```bash
+git config core.hooksPath .githooks
+```
+
+This runs automatically when installing dependencies in the root directory.
+
+When adding a new action, update `.githooks/pre-commit` to include your project:
+
+```bash
+#!/bin/bash
+# set -e exits on error, making `|| exit 1` redundant
+set -euo
+
+# Print each command before executing
+set -x
+
+for project in upstream-tag-on-merge upstream-tag-sync my-new-action; do
+    cd $project
+    echo "Building $project..."
+    bun run build
+    git add dist
+    echo "Linting $project..."
+    bun run lint
+    cd ..
+done
+```
+
+The pre-commit hook:
+
+- Uses bash safety features (`set -euo`)
```

Comment:
IMO it's not necessary to mention this as it's a widely-spread standard directive